### DISCUSSION TITLE
500 Not Found when I try to get the data

### DISCUSSION THREAD

**Discussion by crifano** [kreait/firebase-php#528] [Q&A] [answered]
Good-day to everybody
I use the library version 5.3.0, and until some days ago everything was good
I have this include:

`require_once __DIR__ . '/../vendor/autoload.php';
use Kreait\Firebase\Factory;
use Kreait\Firebase\Database;`
and 
`$dbUri = "https://mycrifanoName.firebaseio.com";
$factory = (new Factory)
	->withServiceAccount(__DIR__ . '/../configure/mycrifanoName-firebase-adminsdk.json')
	->withDatabaseUri($dbUri);
$database = $factory->createDatabase();
$requests = $database->getReference('/requests');
$new_request_ref = $requests->getSnapshot();
`
the last line cause the "crash"
In the log I found this:

`PHP Fatal error:  Uncaught GuzzleHttp\Exception\ConnectException: cURL error 7: Failed to connect to mycrifanoName.firebaseio.com port 443: No route to host (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) in /home/aoespeci/public_html/vendor/guzzlehttp/guzzle/src/Handler/CurlFactory.php:200
Stack trace:
#0 /home/aoespeci/public_html/vendor/guzzlehttp/guzzle/src/Handler/CurlFactory.php(155): GuzzleHttp\Handler\CurlFactory::createRejection(Object(GuzzleHttp\Handler\EasyHandle), Array)
#1 /home/aoespeci/public_html/vendor/guzzlehttp/guzzle/src/Handler/CurlFactory.php(105): GuzzleHttp\Handler\CurlFactory::finishError(Object(GuzzleHttp\Handler\CurlHandler), Object(GuzzleHttp\Handler\EasyHandle), Object(GuzzleHttp\Handler\CurlFactory))
#2 /home/aoespeci/public_html/vendor/guzzlehttp/guzzle/src/Handler/CurlHandler.php(43): GuzzleHttp\Handler\CurlFactory::finish(Object(GuzzleHttp\Handler\CurlHandler), Object(GuzzleHttp\Handler\EasyHandle), Object(GuzzleHttp\Handler\CurlFactory))
#3 /home/aoespeci/public_html/vendor/guz in /home/aoespeci/public_html/vendor/kreait/firebase-php/src/Firebase/Exception/DatabaseApiExceptionConverter.php on line 46
[17-Dec-2020 15:25:33 Europe/Rome] PHP Fatal error:  Uncaught GuzzleHttp\Exception\ConnectException: cURL error 7: Failed to connect to mycrifanoName.firebaseio.com port 443: No route to host (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) in /home/aoespeci/public_html/vendor/guzzlehttp/guzzle/src/Handler/CurlFactory.php:200
`

Until some days ago every was good, and I didn't change nothing
Firebase rules are ok
Any idea? thanks

**Comment by jeromegamez** [comment]
If something worked a couple of days ago and suddenly stopped working, it's probably nothing that's wrong with your code or the SDK. The Firebase API endpoints sometimes are not reachable, but this usually fixes itself after some time.

Just to be sure, please check on https://console.firebase.google.com/project/_/database/ that the database URL is (still) correct.

![rtdb](https://user-images.githubusercontent.com/67554/102684713-64388100-41db-11eb-8cd6-916983b42d39.png)

If the URL shown in the Firebase Console doesn't match what you've used with `withDatabaseUri($dbUri)`, try changing it.

If the URLs do match (that's what I expect), try connecting from another computer and/or another network, to rule out that this is a connectivity issue caused by your environment.

If nothing works, I'd recommend opening a support case with Firebase Support at https://firebase.google.com/support

**Comment by crifano** [comment]
Hi, jerome,
Thanks for your reply 

Unfurtunally every is correct... 
The db url is ok, and already test from other networks 
On plus, in the main time:
- I upgrade the php library to last version but nothing change
- If i make the same call using javascript instead php i can access and write to db without problem

I really do not know what to do, 
Is not good but I thinking to transform my calls on pure js instead using php

**Reply by jeromegamez** [reply]
Could you please try, in your terminal, this command:

```bash
$ curl -I https://mycrifanoName.firebaseio.com
```

This should return something like

```
HTTP/1.1 405 Method Not Allowed
Server: nginx
Date: Sat, 19 Dec 2020 09:43:51 GMT
Content-Type: text/plain
Content-Length: 53
Connection: keep-alive
Access-Control-Allow-Origin: *
Strict-Transport-Security: max-age=31556926; includeSubDomains; preload
```

Could you also tell me which versions of PHP and the kreait libraries you are using? (`php -v` and `composer show | grep kreait`)

**Comment by crifano** [comment]
Yes I have a problem... with curl -I 
I receive 
curl: (7) Failed to connect to 2600:1901:0:94b6::: **Network is unreachable**
Any idea? 

kreait/clock                     1.1.0   A PHP 7.0 compatible clock abstr...
kreait/firebase-php              5.14.0  Firebase Admin SDK
kreait/firebase-tokens           1.14.0  A library to work with Firebase ...

PHP 7.3.25 (cli) (built: Dec  9 2020 03:04:08) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.3.25, Copyright (c) 1998-2018 Zend Technologies
    with the ionCube PHP Loader + ionCube24 v10.4.5, Copyright (c) 2002-2020, by ionCube Ltd.
    with Zend OPcache v7.3.25, Copyright (c) 1999-2018, by Zend Technologies

**Comment by crifano** [comment, marked as answer]
Better... 
If I make the curl command from my computer I have no problem, I receive the same message you show me, but If I make the command from ssh on my server I receve Network is unreachable

**Reply by jeromegamez** [reply]
This does indeed look like an environment issue that's not related to the code. Some hints that might help (but I can only guess):

* Check if the cURL versions on your systems match, and if e.g. the cURL version on your remote system is lower than on your computer, try upgrading cURL. You can check with `curl --version`
* Check that the PHP versions on both systems match; if you can update to PHP 7.4 ([PHP 7.3 is not actively supported anymore](https://www.php.net/supported-versions.php))
* Check the cURL versions bundled with PHP, e.g. by executing `php -r "print_r(curl_version());"` on both systems. It could be that the bundled cURL version on the remote system is somehow buggy.
* Check the Firewall rules on the remote system - it's possible that the domain of your Realtime Database is resolved as an IPv6 address, but outgoing IPv6 connections are prohibited by the Firewall. I believe you could check this with `ping mycrifanoName.firebaseio.com` (for IPv4) and `ping6 mycrifanoName.firebaseio.com` (for IPv6) - the commands could differ depending on the operating system, If it works with the one, but not the other, it's another hint that it might be a Firewall setting

That's all I have at the moment 😅, but if something comes to my mind, I'll add it here. I hope you can find the cause! 🤞

**Reply by crifano** [reply]
Thanks Jerome, the problem seem solved, It was a firewall rule on my server
